src/StreamJsonRpc.Analyzers/SourceWriter.cs

```diff
@@ -0,0 +1,191 @@
+
+/// <summary>
+/// A utility class for generating indented source code.
+/// </summary>
+internal class SourceWriter
+{
+    // Standardize on this because it impacts the roslyn-calculated interceptor magic code,
+    // and we want tests to pass regardless of the platform.
+    private const string NewLine = "\r\n";
+
+    private readonly StringBuilder builder = new();
+
+    private int indentation;
+
+    /// <summary>
+    /// Initializes a new instance of the <see cref="SourceWriter"/> class.
+    /// </summary>
+    public SourceWriter()
+    {
+        this.IndentationChar = '\t';
+        this.CharsPerIndentation = 1;
+    }
+
+    /// <summary>
+    /// Initializes a new instance of the <see cref="SourceWriter"/> class
+    /// with the specified indentation settings.
+    /// </summary>
+    /// <param name="indentationChar">The indentation character to be used.</param>
+    /// <param name="charsPerIndentation">The number of characters per indentation to be applied.</param>
+    public SourceWriter(char indentationChar, int charsPerIndentation)
+    {
+        if (!char.IsWhiteSpace(indentationChar))
+        {
+            throw new ArgumentOutOfRangeException(nameof(indentationChar));
+        }
+
+        if (charsPerIndentation < 1)
+        {
+            throw new ArgumentOutOfRangeException(nameof(charsPerIndentation));
+        }
+
+        this.IndentationChar = indentationChar;
+        this.CharsPerIndentation = charsPerIndentation;
+    }
+
+    /// <summary>
+    /// Gets the character used for indentation.
+    /// </summary>
+    public char IndentationChar { get; }
+
+    /// <summary>
+    /// Gets the number of characters per indentation.
+    /// </summary>
+    public int CharsPerIndentation { get; }
+
+    /// <summary>
+    /// Gets the length of the generated source text.
+    /// </summary>
+    public int Length => this.builder.Length;
+
+    /// <summary>
+    /// Gets or sets the current indentation level.
+    /// </summary>
+    public int Indentation
+    {
+        get => this.indentation;
+        set
+        {
+            if (value < 0)
+            {
+                Throw();
+                static void Throw() => throw new ArgumentOutOfRangeException(nameof(value));
+            }
+
+            this.indentation = value;
+        }
+    }
+
+    /// <summary>
+    /// Appends a single character and then a new line.
+    /// </summary>
+    /// <param name="value">The value to write.</param>
+    public void WriteLine(char value)
+    {
+        this.AddIndentation();
+        this.builder.Append(value);
+        this.builder.Append(NewLine);
+    }
+
+    /// <summary>
+    /// Appends a new line with the specified text.
+    /// </summary>
+    /// <param name="text">The text to append.</param>
+    /// <param name="disableIndentation">Append text without preserving the current indentation.</param>
+    public void WriteLine(
+        [StringSyntax("c#-test")] string text,
```

Comment:
@matteo-prosperi We should add this syntax to our Gladstone configuration object unit tests to get colorization there